src/compiler/scala/tools/nsc/backend/jvm/BTypes.scala

```diff
@@ -891,52 +891,113 @@ abstract class BTypes {
   def isCompilingPrimitive: Boolean
 
   // The [[Lazy]] and [[LazyVar]] classes would conceptually be better placed within
-  // PostProcessorFrontendAccess (they access the `frontendLock` defined in that class). However,
+  // PostProcessorFrontendAccess (they may access the `frontendLock` defined in that class). However,
   // for every component in which we define nested classes, we need to make sure that the compiler
   // knows that all component instances (val frontendAccess) in various classes are all the same,
   // otherwise the prefixes don't match and we get type mismatch errors.
   // Since we already do this dance (val bTypes: GenBCode.this.bTypes.type = GenBCode.this.bTypes)
   // for BTypes, it's easier to add those nested classes to BTypes.
 
-  object Lazy {
-    def apply[T <: AnyRef](t: => T): Lazy[T] = new Lazy[T](() => t)
+  abstract sealed class Lazy[T] {
+    /** get the result of the lazy value, calculation the result and performing the additional actions if the value
+      * is not already known.
+      * @return the
```

Comment:
Let's keep the doc comments concise; the main comment is enough, no need for the annotations (also in the rest of the PR). Also, the comment here is a bit hard to read.

Reply:
done